jwql/jwql_monitors/generate_proposal_thumbnails.py

```diff
@@ -0,0 +1,60 @@
+#! /usr/bin/env python
+
+"""Generate thumbnails for each proposal to be used for display in
+the archive page of the web app.
+
+In the web app, each instrument has its own 'archive' page where users
+may view JWST images for the particular instrument.  The page displays
+buttons for each proposal with a thumbnail image from each particular
+proposal.  This script goes through the thumbnail filesystem and
+creates the thumbnail to display based on the first existing rate file.
+The thumbnail is saved as ``<proposal_id>.thumb``.
+
+Authors
+-------
+
+    - Matthew Bourque
+
+Use
+---
+
+    This script is intended to be executed as such:
+
+    ::
+
+        python generate_proposal_thumbnails.py
+"""
+
+import glob
+import logging
+import os
+import shutil
+
+from jwql.utils.logging_functions import configure_logging, log_info, log_fail
+from jwql.utils.utils import get_config
+
+
+@log_fail
+@log_info
+def generate_proposal_thumbnails():
+    """The main function of the ``generate_proposal_thumbnails`` module.
+    See module docstring for further details."""
+
+    proposal_dirs = glob.glob(os.path.join(get_config()['thumbnail_filesystem'], '*'))
+
+    for proposal_dir in proposal_dirs:
+        rate_thumbnails = glob.glob(os.path.join(proposal_dir, '*rate*.thumb'))
+        if rate_thumbnails:
+            thumbnail = rate_thumbnails[0]
```

Comment:
Is there any case where we would expect the first thumbnail to not be representative of the set of thumbnails? This seems like a very good way to be consistent, though, and I'm not sure that it matters if it's not a representative thumbnail.

Reply:
Great question.  I did not re-work that logic as part of this PR, so using the first `rate` file as a thumbnail is how we currently do it, but perhaps this could be improved.  My hunch is that (at least for now), this is not a big deal, since the thumbnails are really just for aesthetics and are not intended to be used for any sort of meaningful way.

Reply:
Okay, that seems very reasonable!